### PR TITLE
comment fixes

### PR DESCRIPTION
**Pull request by ryan-williams**
- [x] ~Tests added / passed~ (N/A)
- [x] Passes `black dask` / `flake8 dask`

Factored out of #6661 (DDF.iloc).

### CODE REVIEW

**Comment by ryan-williams** [comment]
hmm `dask/array/tests/test_linalg.py::test_norm_any_slice[False--1-shape1-chunks1]`, I suspect this is a flaky test.

Is there a way I can trigger a retry of a given GHA?

**Comment by jrbourbeau** [comment]
Apologies for the inconvenience, `test_norm_any_slice` is a known flaky tests which we're working to resolve (xref https://github.com/dask/dask/issues/7189). I just bumped CI

File: docs/requirements-docs.txt [hunk 1]
```diff
@@ -1,5 +1,5 @@
 numpydoc
-sphinx
+sphinx<3.5.0
```
Comment (jrbourbeau) [inline docs/requirements-docs.txt:2]: Per the discussion over in https://github.com/dask/dask/pull/7217

```suggestion
sphinx
```

**Comment by ryan-williams** [comment]
Thanks! I also rebased this on top of #7217 to try to get CI passing; can re-rebase once #7220 is in

File: dask/dataframe/io/tests/test_io.py [hunk 2]
```diff
@@ -221,6 +221,7 @@ def test_from_bcolz_column_order():
 
 def test_from_pandas_dataframe():
     a = list("aaaaaaabbbbbbbbccccccc")
+    np.random.seed(123)
```
Comment (jrbourbeau) [inline dask/dataframe/io/tests/test_io.py:224]: This should be testing behavior that is independent of the values in the `b` column. Did you find that not having the seed set resulting in this test failing?

File: dask/dataframe/io/parquet/utils.py [hunk 3]
```diff
@@ -462,14 +462,12 @@ def _scrub(i, p):
...
             basepath = basepath[:j]
         l = len(basepath)
-
     else:
         basepath = _join_path(root).split("/")
         l = len(basepath)
         assert all(
             p[:l] == basepath for p in path_parts_list
         ), "All paths must begin with the given root"
-    l = len(basepath)
```
Comment (jrbourbeau) [inline dask/dataframe/io/parquet/utils.py:472]: It looks like `l` is used a few lines down in `"/".join(path_parts[l:])`

File: dask/dataframe/tests/test_dataframe.py [hunk 4]
```diff
@@ -2535,13 +2535,6 @@ def test_drop_columns(columns):
     assert_eq(df.drop(columns=columns), ddf2)
 
 
-def test_gh580():
-    df = pd.DataFrame({"x": np.arange(10, dtype=float)})
-    ddf = dd.from_pandas(df, 2)
-    assert_eq(np.cos(df["x"]), np.cos(ddf["x"]))
-    assert_eq(np.cos(df["x"]), np.cos(ddf["x"]))
```
Comment (jrbourbeau) [inline dask/dataframe/tests/test_dataframe.py:2542]: Why remove this?

**Review by jrbourbeau — COMMENTED** [review]
Thanks for the cleanup @ryan-williams!

File: dask/dataframe/io/parquet/utils.py (hunk 3 above)
Comment (ryan-williams) [reply dask/dataframe/io/parquet/utils.py:472]: yea, it's set in both the `if` and `else` branches above though, so this line is redundant

File: dask/dataframe/io/tests/test_io.py (hunk 2 above)
Comment (ryan-williams) [reply dask/dataframe/io/tests/test_io.py:224]: Yea, agreed that it shouldn't matter here. I added this a while back unfortunately and don't remember the specifics.

My guess is I was debugging general `from_pandas` failures with my iloc implementation, and found it hard to follow when stepping repeatedly in the debugger when the values weren't the same each time.

Happy to remove if it feels like a distraction (or, to file an issue for a more holistic audit that all random tests are seeded, which I feel like should be the default in general)

File: dask/dataframe/tests/test_dataframe.py (hunk 4 above)
Comment (ryan-williams) [reply dask/dataframe/tests/test_dataframe.py:2542]: I started out trying to just add a comment documenting what this test was doing, since it's a bit cryptic, and ultimately decided it was stale/redundant enough to be considered "dead code".

Looking into the history (#580, #582) it seems it was verifying some workaround related to ufuncs that has since been superseded with more correct behavior. I'm not sure if the duplicated assertion is intentional or was ever significant, and I believe e.g. `test_ufunc.py` covers whatever is being tested here (and then some).

Happy to leave it in, but in that case maybe someone else should try their hand at documenting exactly what this test is doing? If no one can, that seems like a good motivation to remove it, I think?

File: docs/requirements-docs.txt (hunk 1 above)
Comment (ryan-williams) [reply docs/requirements-docs.txt:2]: thanks, removed

**Review by ryan-williams — COMMENTED** [review]
thanks, think I responded to everything, will force-push a new rebased version now

File: dask/array/core.py [hunk 5]
```diff
@@ -3544,7 +3544,7 @@ def common_blockdim(blockdims):
 
     if np.isnan(sum(map(sum, blockdims))):
         raise ValueError(
-            "Arrays chunk sizes (%s) are unknown.\n\n"
+            "Array's chunk sizes (%s) are unknown.\n\n"
```
Comment (jsignell) [inline dask/array/core.py:3547]:
```suggestion
            "Arrays' chunk sizes (%s) are unknown.\n\n"
```

**Review by jsignell — COMMENTED** [review]
This PR would be easier to review if it did one thing. Comment improvements would be merged very quickly while discussion of changes to tests could happen separately.

**Comment by ryan-williams** [comment]
alright, I trimmed this to just the comment typo fixes. I'll send a PR w/ the dead code removal, but defer to y'all about whether the "obsolete test" removal / random-test-seeding is worth the PR noise.

**Comment by ryan-williams** [comment]
hmm [this looks like a spurious failure](https://github.com/dask/dask/pull/7215/checks?check_run_id=2108558820)? pip failed to resolve some dep? Can someone try running it again? (I dl'd a full log just in case)

```
Run pre-commit/action@v2.0.0
install pre-commit
  /opt/hostedtoolcache/Python/3.9.2/x64/bin/pip install pre-commit
  Collecting pre-commit
    Downloading pre_commit-2.11.1-py2.py3-none-any.whl (187 kB)
  Collecting virtualenv>=20.0.8
    Downloading virtualenv-20.4.2-py2.py3-none-any.whl (7.2 MB)
  Collecting toml
    Downloading toml-0.10.2-py2.py3-none-any.whl (16 kB)
  Collecting pyyaml>=5.1
    Downloading PyYAML-5.4.1-cp39-cp39-manylinux1_x86_64.whl (630 kB)
  ERROR: Exception:
  Traceback (most recent call last):
    File "/opt/hostedtoolcache/Python/3.9.2/x64/lib/python3.9/site-packages/pip/_vendor/resolvelib/resolvers.py", line 171, in _merge_into_criterion
      crit = self.state.criteria[name]
  KeyError: 'pyyaml'
```

**Review by jrbourbeau — APPROVED** [review]
Thanks @ryan-williams!
